### PR TITLE
feat(android): implemented getSupportedMediaTypeList method

### PR DESCRIPTION
**Pull request by Nikunj-bisht**
## Description

Added `getSupportedMediaTypeList()` that allows to find supported Codecs for device

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅     |
| Windows |    ❌     |

## Checklist

* [X] I have tested this on a device/simulator for each compatible OS
* [X] I added the documentation in `README.md`
* [X] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [X] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -163,7 +163,8 @@ The example app in this repository shows an example usage of every single API, c
 | [getPowerState()](#getpowerstate)                                 | `Promise<object>`   |  ✅  |   ✅    |   ✅    | ✅  |
 | [getProduct()](#getproduct)                                       | `Promise<string>`   |  ❌  |   ✅    |   ❌    | ❌  |
 | [getPreviewSdkInt()](#getPreviewSdkInt)                           | `Promise<number>`   |  ❌  |   ✅    |   ❌    | ❌  |
-| [getReadableVersion()](#getreadableversion)                       | `string`            |  ✅  |   ✅    |   ✅    | ❌  |
+| [getReadableVersion()](#getreadableversion)                       | `string`            |  ✅  |   ✅    |   ✅    | ❌  |## Usage
+
```
Comment (mikehardy) [inline README.md:167]:
```suggestion
```

This change is unrelated to your PR, please do not include unrelated changes

File: README.md [hunk 2]
```diff
@@ -202,6 +203,7 @@ The example app in this repository shows an example usage of every single API, c
 | [supported64BitAbis()](#supported64BitAbis)                       | `Promise<string[]>` |  ❌  |   ✅    |   ❌    | ❌  |
 | [supportedAbis()](#supportedAbis)                                 | `Promise<string[]>` |  ✅  |   ✅    |   ✅    | ❌  |
 | [syncUniqueId()](#syncuniqueid)                                   | `Promise<string>`   |  ✅  |   ❌    |   ❌    | ❌  |
+| [getSupportedMediaTypeList()](#getSupportedMediaTypeList)         | `Promise<string[]>`   |  ❌  |   ✅    |   ❌    | ❌  |
```
Comment (mikehardy) [inline README.md:206]: formatting seems like it has one char of space too many

```suggestion
| [getSupportedMediaTypeList()](#getSupportedMediaTypeList)        | `Promise<string[]>`   |  ❌  |   ✅    |   ❌    | ❌  |
```

**Comment by mikehardy** [comment]
> @mikehardy made the changes mentioned removed all the unrelated changes that was pushed by mistake

With apologies, @Nikunj-bisht that statement is false

This PR adds one method for android only, yet has a diff size of +453 -175

That's outrageous! 😅  Please *just add your implementation of the method and nothing unrelated* then the PR will be ready for review

**Comment by Nikunj-bisht** [comment]
Ok sure @mikehardy let me raise another fresh pr but changes in this pr as you mention 400+ in which some are the formatting changes so that's why 😅

**Comment by mikehardy** [comment]
I understand what the changes are (formatting) I just need you - as a submitter here and potential future submitter to any open source project - that mixing formatting changes in with functional changes is not okay for the reviewer

2 big reasons:

1- formatting is unimportant but any decisions you make on formatting (and then attempt to "enforce" with your commit changing the code to conform to your new formatting style) will likely be wrong, then everyone will argue over this unimportant spacing stuff

2- the diff is huge and the most scarce commodity in the entire open source ecosystem is *maintainers*. We have no time. We have negative time somehow. Burning maintainer time by making them look through unimportant formatting stuff is wasteful, don't do it :-)

I'm looking forward to the clean PR though, as a maintainer I *love* merging clean PRs and releasing them for everyone. Cheers

**Comment by Nikunj-bisht** [comment]
Understood @mikehardy let me create a fresh pr with only required changes

**Comment by Nikunj-bisht** [comment]
@mikehardy i have raised a new pr for same and closing this one.
